### PR TITLE
Space mine!

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR

added space mine! 
image
image

## Why / Balance

image

## Media

https://github.com/space-wizards/space-station-14/assets/96445749/6c16acea-3f80-417d-95fa-d741894ebbac

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
no cl (need add to mapping)

### CODE REVIEW

**Comment by MACMAN2003** [comment]
based and unexploded ordinancepilled

**Comment by Cojoke-dot** [comment]
Gah, I want wait to hear all salv is stuck in space due to hitting a random spawn one of these.

**Comment by LittleNyanCat** [comment]
in the examine text "hanging in the air"... wouldn't it make more sense to be in space?

File: Resources/Prototypes/Entities/Structures/space_mine.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  - type: ExplodeOnTrigger
+  - type: Explosive
+    explosionType: Default
+    maxIntensity: 15
+    intensitySlope: 3
+    totalIntensity: 150
+    canCreateVacuum: true
+  - type: Construction
+    graph: SpaceMine
+    node: start
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/space_mine.yml:58]: Can't this just inherit from basestructuredynamic and remove half the components?

**Comment by TheShuEd** [comment]
- reparent to BaseStructureDynamic
- fix air in description
- add safe mine variant
- add random space mine spawner (spawns dangerous and safe mines) the intrigue - will you die if you touch this mine or not?
- added explosion trigger on contact with a grid

**Comment by WarMechanic** [comment]
this is a good way to punish shittles with no walls while we dont have grid collision damage

**Comment by ps3moira** [comment]
Would you be able to make it so there's a somewhat small chance it won't explode? I mean how long can space mines last right?

**Comment by TheShuEd** [comment]
Already added safe mine, check comments

File: Content.Server/Explosion/EntitySystems/TriggerSystem.cs [hunk 2]
```diff
@@ -250,6 +252,14 @@ private void OnRepeatInit(Entity<RepeatingTriggerComponent> ent, ref MapInitEven
...
         }
 
+        private void OnGridCollide(Entity<TriggerOnGridCollideComponent> ent, ref EntParentChangedMessage args)
+        {
+            if (!TryComp<MapGridComponent>(Transform(ent).ParentUid, out var grid))
+                return;
+
+            Trigger(ent);
+        }
+
```
Comment (metalgearsloth) [inline Content.Server/Explosion/EntitySystems/TriggerSystem.cs:262]: How come it's not using a collision event with the grid collision mask?

File: Content.Server/Explosion/Components/TriggerOnGridCollideComponent.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace Content.Server.Explosion.Components
+{
+    [RegisterComponent]
+    public sealed partial class TriggerOnGridCollideComponent : Component
+    {
+    }
+}
```
Comment (metalgearsloth) [inline Content.Server/Explosion/Components/TriggerOnGridCollideComponent.cs:7]: 1. File-scoped namespace.
2. Shared.
3. Needs documenting.

**Comment by ps3moira** [comment]
Are you still planning to finish this?

**Comment by TheShuEd** [comment]
yep

**Comment by TheShuEd** [comment]
sorry, I didn't get a chance to test working with the grid collider, I don't want to finish this. Use the sprites as you wish.

**Comment by dvir001** [comment]
@TheShuEd You cool with me reworking this for frontier and marking it as MIT?

**Comment by TheShuEd** [comment]
> @TheShuEd You cool with me reworking this for frontier and marking it as MIT?

It's cool, do it

**Pull request closed without merging**